api/tests/MemberMatch/MemberMatchTests.cs

```diff
@@ -326,5 +329,156 @@ public void TestMethod9()
+                from member in candidateMembers
+                let score = ScoreMember(scorer, cityToFrequency, member, tokenizedLine)
+                where score > minimumScore
+                orderby score descending
+                select (member, score)).ToList();
+
+            foreach (var (member, score) in memberAndScoreList)
+            {
+                Debug.WriteLine($"   {score:0.00} : {member}");
+                Debug.WriteLine("TODO remove");
+            }
+
+            return memberAndScoreList;
+        }
+
+        private static (string, HashSet<string>) TokenizedLine(string line)
+        {
+            //!!!TODO split on all whitespace, too.
+            string[] separatingStrings = { ".", ",", " ", "\t" };
+
+            string upper_line = line.ToUpperInvariant();
+
+            var tokenSet = (from token in upper_line.Split(separatingStrings, System.StringSplitOptions.RemoveEmptyEntries)
+                            from token2 in Member.ProcessName(token)
+                            select token2)
+                            .ToHashSet();
+
+            foreach (var token in tokenSet)
+            {
+                Debug.WriteLine($"  > '{token}'");
+            }
+
+            return (upper_line, tokenSet);
+        }
+
+        private static double ScoreMember(Scorer scorer, Dictionary<string, double> cityToFrequency, Member member, (string, HashSet<string>) tokenizedLine)
+        {
+            var (upper_line, tokenSet) = tokenizedLine;
+
+            var firstNameList = member.FirstList;
+            var firstNameIsContainedList = (
+                from first in member.FirstList
+                select tokenSet.Contains(first))
+                .ToArray();
+            var lastNameList = member.LastList;
+            var lastNameIsContainedList = (
+                from last in member.LastList
+                select tokenSet.Contains(last))
+                .ToArray();
+
+            var score = Scorer.DefaultPriorScore;
+            score += scorer.Delta(
+                                 nameList: firstNameList,
+                                 isContainedList: firstNameIsContainedList);
+            score += scorer.Delta(
+                                 nameList: lastNameList,
+                                 isContainedList: lastNameIsContainedList);
+            if (cityToFrequency != null)
+            {
+                double cityFrequency = cityToFrequency[member.City];
+                bool cityIsContained = upper_line.Contains(member.City); // TODO: This test function appears in two places and could get out of sync
+                score += Scorer.Delta(probabilityAppearsInLineByCoincidence: cityFrequency, isContained: cityIsContained);
+            }
+
+            return score;
+        }
+
+        private Scorer TestScorer()
+        {
+            var nameToProbability = new Dictionary<string, double>()
+            {
+                { "JOHN", .01 },
```

Comment:
where are these numbers coming from?

Reply:
These are just made up (but roughly correct). They allow the tests to run without needing any file or database access. The tests make sure that any changes to this part of the code won't unexpectedly change the results.